### PR TITLE
[SPARK-24197][SparkR][FOLLOWUP] Fixing failing tests for array_sort and sort_array

### PR DESCRIPTION
**Pull request by mn-mikke**
## What changes were proposed in this pull request?

The PR tries to fix [the problem](https://github.com/apache/spark/pull/21294#discussion_r189063233) with failing tests for the SparkR functions `array_sort` and `sort_array`.

### CODE REVIEW

**Comment by mn-mikke** [comment]
cc @felixcheung @HyukjinKwon @gatorsmile

**Comment by ueshin** [comment]
Hi, any updates on this?

**Comment by mn-mikke** [comment]
@ueshin Have you experienced the same problem with the failing tests?

**Comment by felixcheung** [comment]
We haven’t seen it fail?

**Comment by ueshin** [comment]
@mn-mikke Actually, yes. But unfortunately I'm not familiar with R and the environment I heard about it, so I'm not able to figure out the reason for now.

**Comment by gatorsmile** [comment]
Will do more investigation. Thanks!

File: R/pkg/tests/fulltests/test_sparkSQL.R [hunk 1]
```diff
@@ -1504,15 +1504,16 @@ test_that("column functions", {
   expect_equal(result, "cba")
 
   # Test array_sort() and sort_array()
-  df <- createDataFrame(list(list(list(2L, 1L, 3L, NA)), list(list(NA, 6L, 5L, NA, 4L))))
+  df <- createDataFrame(list(list(list(2L, 1L, 3L, NA)), list(list(NA, 5L, NA, 4L))))
+  as_integer_lists <- function(x) lapply(x, lapply, as.integer)
 
-  result <- collect(select(df, array_sort(df[[1]])))[[1]]
-  expect_equal(result, list(list(1L, 2L, 3L, NA), list(4L, 5L, 6L, NA, NA)))
+  result <- as_integer_lists(collect(select(df, array_sort(df[[1]])))[[1]])
```
[the remark this answers is not in the document]
Comment (HyukjinKwon) [reply R/pkg/tests/fulltests/test_sparkSQL.R]: +1

**Comment by HyukjinKwon** [comment]
@mn-mikke, shall we leave this closed until we have the root cause?

**Pull request closed without merging**